lib/fusion_auth/openid_connect.ex

```diff
@@ -0,0 +1,223 @@
+defmodule FusionAuth.OpenIdConnect do
+  @moduledoc """
+  The `FusionAuth.OpenIdConnect` module provides access methods to the [FusionAuth OpenID Connect API](https://fusionauth.io/docs/v1/tech/apis/identity-providers/openid-connect).
+  All methods require a Tesla Client struct created with `FusionAuth.client(base_url, api_key, tenant_id)`.
+  """
+
+  @type client :: FusionAuth.client()
+  @type result :: FusionAuth.result()
+  @type domain :: String.t()
+  @type headers :: list()
+  @type identity_provider_id :: String.t()
+  @type application_id :: String.t()
+  @type tenant_id :: String.t()
+  @type identity_provider :: %{
+    application_configuration: %{
+      application_id() => application_configuration()
+    },
+    button_image_url: String.t(),
+    button_text: String.t(),
+    debug: boolean(),
+    domains: [String.t()],
+    lambda_configuration_reconciled: String.t(),
+    linking_strategy: String.t(),
+    name: String.t(),
+    oauth_configuration: oauth_configuration(),
+    post_request: boolean(),
+    tenant_configuration_map: %{
+      tenant_id() => tenant_configuration()
+    },
+    type: String.t()
+  }
+  @type application_configuration :: %{
+    button_image_url: String.t(),
+    button_text: String.t(),
+    oauth_client_id: String.t(),
+    oauth_client_secret: String.t(),
+    create_registration: boolean(),
+    enabled: boolean(),
+    oauth_scope: String.t()
+  }
+  @type oauth_configuration :: %{
+    authorization_endpoint: String.t(),
+    client_id: String.t(),
+    client_secret: String.t(),
+    client_authentication_method: String.t(),
+    email_claim: String.t(),
+    issuer: String.t(),
+    scope: String.t(),
+    token_endpoint: String.t(),
+    userinfo_endpoint: String.t(),
+  }
+  @type tenant_configuration :: %{
+    limit_user_link_count: %{
+      enabled: integer(),
+      maximum_links: integer(),
+    }
+  }
+
+  @type login_request_body :: %{
+    application_id: String.t(),
+    identity_provider_id: String.t(),
+    ipAddress: String.t(),
+    data: %{
+      code: String.t(),
+      redirect_uri: String.t(),
+    },
+    metadata: %{
+      device: %{
+        description: String.t(),
+        lastAccessedAddress: String.t(),
+        name: String.t(),
+        type: String.t(),
+      }
+    },
+    no_jwt: boolean(),
+    no_link: boolean()
+  }
+  @connect_url "/api/identity-provider"
+
+  @doc """
+  Retrieves all identity providers
+
+  For more information, visit the FusionAuth API documentation for [Retrieve all identity providers](https://fusionauth.io/docs/v1/tech/apis/identity-providers/#retrieve-all-identity-providers)
+  """
+  @spec retrieve_all_identity_providers(client()) :: FusionAuth.result()
```

Comment:
Since you defined `@type result :: FusionAuth.result()` above, you could replace the `:: FusionAuth.result()` portion here with just `:: result()`. It's up to you, just consistency one way or another is ideal. Since we're already specifying the `client()` instead of `FusionAuth.client()` in the param, makes sense to just make them consistent one way or another